climada/hazard/landslide.py

```diff
@@ -188,304 +127,170 @@ def __init__(self):
-        pixel_width = raster.meta['transform'][0]
-        pixel_height = raster.meta['transform'][4]
+        ls_gdf_bbox = gpd.read_file(path_sourcefile, bbox=bbox)
+        
+        self.centroids.set_raster_from_pnt_bounds(bbox,res)
+
+        n_ev = len(ls_gdf_bbox)
+        
+        # assign lat-lon points of LS events to corresponding grid & flattened
+        # grid-index
+        ls_gdf_bbox = ls_gdf_bbox.reindex(columns = ls_gdf_bbox.columns.tolist() + 
+                                          ['col','row'])
+        ls_gdf_bbox[['col', 'row']] = ls_gdf_bbox.apply(
+            lambda row: mapping_point2grid(row.geometry, bbox[-1],bbox[0], 
+                                           res),
+            axis = 1).tolist()
+        
+        ls_gdf_bbox['flat_ix'] = ls_gdf_bbox.apply(
+            lambda row: mapping_grid2flattened(row.col, row.row, 
+                                               self.centroids.shape), 
+            axis = 1)
+        
+        self.intensity = sparse.csr_matrix(
+            (np.ones(n_ev),(np.arange(n_ev),ls_gdf_bbox.flat_ix)), 
+            shape=(n_ev, self.centroids.size))
+        self.units = 'm/m'
+        self.event_id = np.arange(n_ev, dtype=int)
+        self.orig = np.ones(n_ev, bool)
+        if hasattr(ls_gdf_bbox, 'ev_date'):
+            self.date = pd.to_datetime(ls_gdf_bbox.ev_date, yearfirst=True)
+        else:
+            LOGGER.info('No event dates set from source')
+        if not self.date.empty:
+            self.frequency = np.ones(n_ev)/(
+                (self.date.max()-self.date.min()).value/3.154e+16)
+        else: 
+            LOGGER.info('No frequency can be derived, no event dates')
+        self.fraction = self.intensity.copy()
 
-        return pixel_height, pixel_width
+        self.check()
 
-    def _intensity_cat_to_prob(self, max_prob):
-        """convert NASA nowcasting categories into occurrence probabilities:
-            highest value category value receives a prob of max_prob, lowest category value
-            receives a prob value of 0"""
-        self.intensity_cat = self.intensity.copy()  # save prob values
-        self.intensity = self.intensity.astype(float)
-        self.intensity.data = self.intensity.data.astype(float)
-        max_value = float(max(self.intensity_cat.data))
-        min_value = float(min(self.intensity_cat.data))
+        if check_plots:
+            self.centroids.plot()
+            self.plot_intensity(0)
 
-        for i, j in zip(*self.intensity.nonzero()):
-            self.intensity[i, j] = float((self.intensity[i, j] - min_value) /
-                                         (max_value - min_value) * max_prob)
+        return self
 
 
-    def _intensity_prob_to_binom(self, n_years):
-        """convert occurrence probabilities in NGI/UNEP landslide hazard map into binary
-        occurrences (yes/no) within a given time frame.
+    def set_ls_prob(self, bbox, path_sourcefile, corr_fact=10e6, n_years=500, 
```

Comment:
Maybe the naming of this method should reflect the fact that it is only compatible with the UNEP & NGI dataset.